security/pkg/pki/ra/k8s_ra.go

```diff
@@ -81,23 +114,29 @@ func (r *KubernetesRA) kubernetesSign(csrPEM []byte, caCertFile string, certSign
 	}
 	certChain, _, err := chiron.SignCSRK8s(r.csrInterface, csrPEM, certSigner, usages, "", caCertFile, true, false, requestedLifetime)
 	if err != nil {
+		log.Infof("K8S RA error %s %v", certSigner, err)
 		return nil, raerror.NewError(raerror.CertGenError, err)
 	}
 	return certChain, err
 }
 
 // Sign takes a PEM-encoded CSR and cert opts, and returns a certificate signed by k8s CA.
+// Should return the full chain (tls.crt style) - we don't have the intermediaries.
+// The roots are returned from the normal files (same as Istiod's own trusted roots).
 func (r *KubernetesRA) Sign(csrPEM []byte, certOpts ca.CertOpts) ([]byte, error) {
-	_, err := preSign(r.raOpts, csrPEM, certOpts.SubjectIDs, certOpts.TTL, certOpts.ForCA)
+	ttl, err := preSign(r.raOpts, csrPEM, certOpts.SubjectIDs, certOpts.TTL, certOpts.ForCA)
 	if err != nil {
 		return nil, err
 	}
-	certSigner := certOpts.CertSigner
 
-	return r.kubernetesSign(csrPEM, r.raOpts.CaCertFile, certSigner, certOpts.TTL)
+	// certOpts.CertSigner will never be set - Sign is only called if certSigner is empty
+	// Passing anything else as certSigner will break because kubernetesSign checks if domain is set.
+	return r.kubernetesSign(csrPEM, r.raOpts.CaCertFile, "", ttl)
```

Comment:
Sign is called from `SignWithCertChain` which is called from `CreateCertificate` when certSigner is not empty.